### PR TITLE
feat(appengine): added a new sample 'static-files-spa' 

### PR DESCRIPTION
**Pull request by evil-shrike**
The app demonstrates static files hosting in SPA scenarios.

Supposed to be used as a sample for new section in for https://cloud.google.com/appengine/docs/standard/nodejs/serving-static-files

### CODE REVIEW

File: appengine/static-files-spa/app.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const app = express();
+
+const STATIC_DIR = process.env.STATIC_DIR || 'public';
+const staticFilesDir = path.join(__dirname, STATIC_DIR);
+// Use the built-in express middleware for serving static files
+app.use(express.static(staticFilesDir));
+
+// All api routes go here
+
+// support of client-side routing in SPA apps (all routes that are not api lead to root)
```
Comment (ace-n) [inline appengine/static-files-spa/app.js:29]: Nit
```suggestion
// Support client-side routing in SPA apps (all routes that other than `/api` lead to root)
```

File: appengine/static-files-spa/public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <title>Static Files</title>
+  <base href="/">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="icon" type="image/x-icon" href="favicon.ico">
+  <link rel="stylesheet" href="main.css">
+  <script type='text/javascript'>
+
```
Comment (ace-n) [inline appengine/static-files-spa/public/index.html:26]: Nit: delete empty line.

File: appengine/static-files-spa/public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  function onNavigate(page) {
```
Comment (ace-n) [inline appengine/static-files-spa/public/index.html:27]: Nit: indent the contents of the `<script>...</script>` block by 2 more spaces.

File: appengine/static-files-spa/public/index.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    window.history.pushState({urlPath: page},"",page)
+    console.log('Navigated to ' + page);
+    return false;
+  }
+  </script>
+</head>
+<body>
+  <p>This is a static file serving example with SPA support.</p>
+  <p>Mind the absence of page reloads on activating pages via the menu</p>
```
Comment (ace-n) [inline appengine/static-files-spa/public/index.html:36]:
```suggestion
<p>Clicking links in the menu will <em>not</em>cause a page reload.</p>
```

**Review by ace-n — CHANGES_REQUESTED** [review]
LGTM other than a few nits.

I'm also going to clarify who should own this sample internally with my immediate team.

**Comment by evil-shrike** [comment]
@ace-n thanks, updated the code with your corrections

**Comment by dinagraves** [comment]
Waiting for resolution on sample: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/6727#issuecomment-1023561960

**Comment by ace-n** [comment]
The [companion Python PR](https://github.com/GoogleCloudPlatform/python-docs-samples/pull/6727#issuecomment-1040602712) was closed, so I'm closing this one as well.

**Pull request closed without merging**